File: programmers/js/level01/132267.js
// 콜라 문제
function solution(a, b, n) {
  let answer = 0;
  let bottle = n;

  while (bottle >= a) {
    const tmp = ~~(bottle / a) * b;
    bottle = tmp + (bottle % a);
    answer += tmp;
  }

  return answer;
}

// 다른 풀이
// function solution(a, b, n) {
//   return ~~((n - b) / (a - b)) * b;
// }

File: programmers/java/level01/132267.java
// 콜라 문제

class Solution {
    public int solution(int a, int b, int n) {
        int result = 0;
        
        while (n >= a) {
            int coke = (int) n / a * b;
            result += coke;
            n = coke + n % a;
        }
        
        return result;
    }
}